app/scripts/utils/functional.es6

```diff
@@ -81,3 +108,11 @@ Result.mapObj = (fn, obj) => reduceResult(fn, (a, b) => a.concat(b), [], asArray
  * @returns {Result<object>} The merged object in a Result.Ok
  */
 Result.merge = (data) => reduceResult((v) => Ok(v), mergeTwo, {}, data);
+
+/**
+ * @param {array<object>} data Some objects to be aggregated and wrapped in Result.Ok
+ * @param {string} aggProp the property name to aggregate
+ * @param {string} countProp the property name to count
+ * @returns {Result<object>} The object in a Result.Ok
+ */
+Result.aggregate = (data, aggProp, countProp) => reduceResult(checkData, (agg, item) => sumAggregate(agg, item, aggProp, countProp), {keys: {aggProps: [], sumProps: []}}, data);
```

Comment:
There is a lot going on here, we should break it into some pieces.

The second argument of `reduceResult` is a reduce function. So it looks like that's what this is doing, but it's complicated enough that it can be its own function. Just writing it out here so I understand:

```js
function sumAggReducer(agg, next) {
  return sumAggregate(agg, item, aggProp, countProp);
}
```

So that actually doesn't work because `aggProp` and `countProp` come from a scope we lost.

hmm.

Actually, stepping back I think this might be doing too much. Aggregating one prop, and summing another prop, sounds like two different tasks.